src/window/empire.c

```diff
@@ -495,56 +538,6 @@ static int is_mouse_hit(tooltip_context *c, int x, int y, int size)
     return x <= mx && mx < x + size && y <= my && my < y + size;
 }
 
-static int get_tooltip_resource(tooltip_context *c)
```

Comment:
Removing this code was sort of a "win", complexity-wise. The `generic_button` input handling knows which resource button was focused, if any. The tooltip code can then retrieve the focused resource type and display that text directly.

Reply:
This is back again to handle the tooltips for a closed city only.